vitrina/resources/models.py

```diff
@@ -66,5 +69,23 @@ class DatasetDistribution(models.Model):
     comment = models.TextField(blank=True, null=True)
 
     class Meta:
-        managed = True
         db_table = 'dataset_distribution'
+
+    def extension(self):
+        if self.filename:
+            name, extension = os.path.splitext(self.filename.name)
+            return extension.replace(".", "").upper()
+        return ""
+
+    def filename_without_path(self):
+        return os.path.basename(self.filename.name) if self.filename else ""
+
+    def get_download_url(self):
+        return reverse('dataset-distribution-download', kwargs={
+            'organization_kind': self.dataset.organization.kind if self.dataset and self.dataset.organization else None,
+            'organization_slug': self.dataset.organization.slug if self.dataset and self.dataset.organization else None,
+            'dataset_slug': self.dataset.slug if self.dataset else None,
```

Comment:
```suggestion
            'dataset_slug': self.dataset.slug,
```

Dataset must always be set.